### ISSUE TITLE
Split and Organize Checkstyle inputs by Test

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#3891]
Taken from https://github.com/checkstyle/checkstyle/pull/3887#discussion_r103765963

It makes it hard to review changes to one Check when multiple Checks are using the same input file and users modify that one file. It makes sense, imo, to add new stuff to an existing input file for simple fixes instead of always creating a new one.

We already restrict inputs to only be used by tests in a single folder. We should take this further and restrict inputs to a single test so no 2 tests use the same input file.

**Each PR should be one single Test class and it's inputs for quicker review, unless an admin has specified otherwise.**

We should start by renaming all inputs to `Input[SimpleTestName].*` (`[]`s are just for separation and should not be in final file name),  that are used by a test. If test is named `ParenPadCheckTest`, simple name would be `ParenPad` and input would start with `InputParenPad`. Any text can be appended to the end to make the input file name unique among others for the same test.

Each test's inputs should be alone in their own directory. If Test was in directory `.../checks/design` then inputs should be located in `.../checks/design/[SimpleTestName]` (`[]`s are just for separation and should not be in final file name). Folder names should be in all lowercase.
Files should be moved to their new locations unless they are used by other checks. If that is the case, then the files should be copied over until all tests have their inputs separated. No input should be left without a test.

We expect no changes to the test's expected violation positions (line or column number). If that does happen because of any of the requested changes, then some of the changes need to be undone.
If violation changes because of class rename, then class name and file name must be left unchanged but package rename and moving to new folder can still be done.
These type of problems will be handled in the final commits.

Once all inputs are renamed, we should see if we can use sevntu's Test to verify all current and future inputs are named after their tests and we have no isolated inputs. https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/sevntu-checks/src/test/java/com/github/sevntu/checkstyle/internal/AllChecksTest.java#L111

**Comment by MEZk** [comment]
@rnveach 
> We should start by renaming all inputs to Input[SimpleTestName].* that are used by a test. If test is named ParenPadCheckTest, simple name would be ParenPad. Any text can be appended to the end to make the input file name unique among others for the same test.

Very good idea as for me. It is really annoying to read a huge input file for a test especially when there is a mixed content (two check's options use the same file, etc).

By the way, if we have individual input file for each test, we can add ```//violation``` comment to make the test input easy to understand and quick to read.

**Comment by rnveach** [comment]
Small update to issue to describe that `[]`s are just for easy name separation and shouldn't be in final file name.
Small change that all inputs should be in their own directory separated from every other test in same package.

@MEZk 
> lets check and update input files for ITs too.

definately.

> By the way, if we have individual input file for each test, we can add //violation comment to each file to make the test input easy to understand and quick to read.

Originally I thought this was ok, but the downside to this is we won't have the same input file for different runs with different properties of the check. We could see something by turning properties on/off on same file.
My only counter ideas are either placing a special violation message distinguishing which test is causing the violation or make on/off files special and kept in-sync by CI.

**Comment by romani** [comment]
lets do split inputs for now only.

upgrade for UTs to describe configuration in comments and have strict "//violation" location is good approach that we already practicing, but lets do this as separate step - separate issue.

**Comment by Kietzmann** [comment]
@rnveach @romani 
Could you please specify which packages(classes) left to move their resources?
Thanks

**Comment by romani** [comment]
@Kietzmann , before sending PR, please run "mvn verify" on your local it will economize you bunch of time.

**Comment by romani** [comment]
@Kietzmann , thanks a lot for desire to help
while I am thinking how to answer on your question, please help us to finish following PRs and close related issues:
#3984
#3971
#3947

I closed them only by timeout, as authors are not responding for long time.

**Comment by Kietzmann** [comment]
@romani 
Seems that I fixed issue described in PR #4068 and issue #4043

**Comment by romani** [comment]
@Kietzmann ,

confirmed, thanks a lot.

Criterias to find what is not done:
1)
Checks that are not covered could be found by simple pattern (taken form ArrayTypeStyleCheckTest)
```
public classXXXXXXXXCheckTest
    extends BaseCheckTestSupport {
    @Override
    protected String getPath(String filename) throws IOException {
        return super.getPath("checks" + File.separator + filename);
    }
```

but required form is 
`return super.getPath("checks" + File.separator + "somecheckname"+ File.separator + filename);`

the same is true for nonCheck - modules.

2)
There should be no input files that are located in non specific to Check/Module folder.
Example : https://github.com/checkstyle/checkstyle/tree/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks
All files in this folder should be in special folder. Check/Module should completely control/own whole his test input folder.

**Comment by romani** [comment]
@rnveach , do you know what else is left to do?

**Comment by rnveach** [comment]
@romani I have 3 branches queued up, `CheckerTest`, `TreeWalkerTest`, `JavadocDetailNodeParser`. After that `grammar`, `astprinter`, `internal` folders have to be done.
Once everything is done, we need to create a UT to control all this so we don't have to keep this in check manually. This will most likely catch anything we did miss,

**Comment by rnveach** [comment]
Any file organization that is left over is split into the 2 issues above.
All that is left in this issue is to create the UT that enforces our naming scheme and fix any other remaining files we missed.

**Comment by romani** [comment]
@rnveach , I recommend to close this issue and create separate issue for UT.
issues with long history are scary.

**Comment by rnveach** [comment]
@romani after #5120 is merged, I only have 1 PR left which is the UT and the final fixes.
Then this issue can be closed.

**Comment by romani** [comment]
all fixes are merged, if smth is left, should be separate issue(s)
